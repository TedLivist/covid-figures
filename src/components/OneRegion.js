import React from 'react';
import PropTypes from 'prop-types';

const OneRegion = (props) => {
  const { name, cases, num } = props;

  return (
    <div className={num % 2 === 0 ? 'one-region d-flex region-even' : 'one-region d-flex region-odd'}>
      <div>{name}</div>
      <div className="d-flex">
        <div>{cases}</div>
        <div className="bx-icon-2">
          <box-icon color="#fff" size="1rem" name="right-arrow-circle" />
        </div>
      </div>
    </div>
  );
};

OneRegion.propTypes = {
  name: PropTypes.string,
  cases: PropTypes.number,
  num: PropTypes.number,
};

OneRegion.defaultProps = {
  num: 0,
  cases: 0,
  name: '',
};

export default OneRegion;
