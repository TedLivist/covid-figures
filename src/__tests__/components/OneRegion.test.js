import React from 'react';
import renderer from 'react-test-renderer';
import OneRegion from '../../components/OneRegion';

test('renders correctly', () => {
  const tree = renderer
    .create(<OneRegion />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})