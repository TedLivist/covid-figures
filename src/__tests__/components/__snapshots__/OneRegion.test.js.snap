// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="one-region d-flex region-odd"
>
  <div />
  <div
    className="d-flex"
  >
    <div />
    <div
      className="bx-icon-2"
    >
      <box-icon
        color="#fff"
        name="right-arrow-circle"
        size="1rem"
      />
    </div>
  </div>
</div>
`;
